### PR TITLE
[dev-launcher] update RN app

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

A companion PR to the `expo-dev-client-components` PR, this one updates the `dev-launcher` app to match the new designs. 

# Test Plan

- added unit tests to ensure the react app works

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by lukmccall — COMMENTED** [review]
I didn't have time today to walk through that code in details (I will do it tomorrow), but generally looks good 🚀

File: packages/expo-dev-launcher/package.json
Comment (ajsmth) [reply packages/expo-dev-launcher/package.json:51]: yea its being used by react-navigation - will look into vendoring it today

**Comment by ajsmth** [comment]
I will be stripping out parts of this as I remove the vendored `react-native-svg`

File: packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts
Comment (esamelson) [inline packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts:41]: i'm a little confused about what's going on here 🤔 do you mean to be setting result.type instead?

File: packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts
Comment (esamelson) [inline packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts:43]:
```suggestion
    result = { type: 'dismiss' };
```

File: packages/expo-dev-launcher/bundle/native-modules/DevMenuInternal.ts
Comment (esamelson) [inline packages/expo-dev-launcher/bundle/native-modules/DevMenuInternal.ts:13]: is this still a TODO?

**Review by esamelson — COMMENTED** [review]
Haven't finished looking through everything yet, but here's a batch of comments to start. Everything that I've looked at so far generally looks great!

File: packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts:41]: sorry these files are copied from the DevMenu package - I haven't really looked into them much yet

File: packages/expo-dev-launcher/bundle/native-modules/DevMenuInternal.ts
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/native-modules/DevMenuInternal.ts:13]: Yes, although I'm considering just including this stuff in this PR instead of the third (draft) one I have open

File: packages/expo-dev-launcher/bundle/screens/SettingsScreen.tsx
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/screens/SettingsScreen.tsx:62]: We could remove it but it's in anticipation of the overlay feature, Im not sure what the status of that is though

File: packages/expo-dev-launcher/bundle/screens/SettingsScreen.tsx
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/screens/SettingsScreen.tsx:118]: yes this is still TODO - sorry!

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt
Comment (lukmccall) [inline packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt:160]:
```suggestion
    map.apply {
      putString("appVersion", packageInfo.versionName)
      putString("appName", appName.toString())
      putString("appIcon", "")
      putString("EXUpdatesRuntimeVersion","")
      putString("EXUpdatesSDKVersion", "")
    }
```

File: packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/DevLauncherActivity.kt
Comment (lukmccall) [inline packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/DevLauncherActivity.kt:47]:
```suggestion
```

File: packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/DevLauncherActivity.kt
Comment (lukmccall) [inline packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/DevLauncherActivity.kt:59]: It is not needed anymore?

File: packages/expo-dev-launcher/bundle/hooks/useDevMenuSettings.tsx
Comment (lukmccall) [inline packages/expo-dev-launcher/bundle/hooks/useDevMenuSettings.tsx:1]: I see what you want to do here, but I think we can do it better. Your current solution looks like `expo-dev-launcher` is depending on `expo-dev-menu`.  We can get settings via `expo-dev-menu-interface` and reexported functions through the dev-launcher internal module (if dev-menu is present). I have one more idea. Maybe we could set `expo-dev-menu` as an optional dependency of the `expo-dev-launcher and export functions to edit setting through the dev menu public module. What do you think? Of course, we can land it, and later we can do the refactor.

File: packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts
Comment (lukmccall) [reply packages/expo-dev-launcher/bundle/native-modules/DevMenuWebBrowser.ts:41]: I know that it looks very hacky and indeed it is. Btw. this code was mostly copied from `expo-web-browser`.

File: packages/expo-dev-launcher/ios/ReactNative/EXDevLauncherRCTBridge.m
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/ReactNative/EXDevLauncherRCTBridge.m:33]:
```suggestion
  NSArray<NSString *> *allowedModules = @[@"RCT", @"DevMenu"];
```

**Review by lukmccall — COMMENTED** [review]
A couple of comments ;) 
Generally, looks very good and solid :D

File: packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/DevLauncherActivity.kt
Comment (ajsmth) [reply packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/DevLauncherActivity.kt:57]: I was having some initial layout issues w/ dev launcher, I think this has something to do with it. also I don't think splash screen is used on iOS at all so I figure it would be a good idea to strip this out and rethink splash at some point

File: packages/expo-dev-launcher/bundle/components/redesign/UrlDropdown.tsx
Comment (esamelson) [inline packages/expo-dev-launcher/bundle/components/redesign/UrlDropdown.tsx:14]: maybe worth at least adding a linear task to figure out later

**Review by esamelson — CHANGES_REQUESTED** [review]
a few more comments on the native side now that there is more stuff there 😄 the JS side looks great for the most part!

File: packages/expo-dev-launcher/bundle/hooks/useAppInfo.tsx
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/hooks/useAppInfo.tsx:17]: yup makes sense!

File: packages/expo-dev-launcher/ios/EXDevLauncherInternal.m
Comment (ajsmth) [reply packages/expo-dev-launcher/ios/EXDevLauncherInternal.m:111]: agreed!

File: packages/expo-dev-launcher/bundle/functions/getLocalPackagersAsync.ts
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/functions/getLocalPackagersAsync.ts:24]: if the fetch call fails for whatever reason it's caught and ignored - Im not sure if that answers your question or not - I haven't noticed any hanging in personal use but it might be worth looking at it a bit harder if you think this might be bug prone!

File: packages/expo-dev-launcher/ios/EXDevLauncherController.m
Comment (ajsmth) [reply packages/expo-dev-launcher/ios/EXDevLauncherController.m]: hmm for some reason this keeps giving me null, whereas the self-constructed url seems to work - I tried specifying a subdirectory for this API as well with no luck - any ideas?

Comment (esamelson) [reply packages/expo-dev-launcher/ios/EXDevLauncherController.m]: ah, I see, it looks like the ways the OS provides to access the icon give you UIImage objects rather than a URL. this is probably fine then.

**Review by esamelson — APPROVED** [review]
I think this looks great. The one thing I'd want to make sure is resolved before landing is the question around the splash screen, as I don't have a clear sense of the implications of commenting out that code. Otherwise we're good to go ahead and land, from my end (once the last few dev URL changes are reverted of course 🙂 ) !

**Review by lukmccall — APPROVED** [review]
I fully agree with Eric. So please, revert those(^^) changes and in my option, we can land it.

File: packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts:36]: I missed this thank you! will update throughout JS

**Comment by ajsmth** [comment]
@lukmccall is this e2e expected to pass? seems to timeout every time

**Comment by lukmccall** [comment]
> is this e2e expected to pass? seems to timeout every time

Somethings broke it and I don't know what 🤷 
You can ignore that for now.

**Pull request merged**
